### ISSUE TITLE
Интересная задача

### ISSUE THREAD

**Issue by SpiritUrban** [robespier/IML_CutLayout#29]
## Готов присоеденится.

У меня хорошо работает геометрическое мышление и получается выводить формулы.

Мой скайп: Spirit_Urban

PERSONAL INFORMATION

Dyachuk Vitaliy
Khmelnitsky (Ukraine) 
Sex Male | Date of birth 03/06/1983 | Nationality Ukrainian 

EDUCATION AND TRAINING 

1999–2001 Skilled worker
Higher Vocational School, Khmelnytsky (Ukraine) electrician on repairand maintenance of electrical equipment

2001–2005 Specialist 
Kamenetz-Podolsk State University, Kamenetz-Podolsk (Khmelnytsky) (Ukraine) 
Teacher of Physical Education, Valeology, Organizer of tourism work

WORK EXPERIENCE 

2005-2006 Teacher of Physical Education
2005-2009 Martial arts trainer
2006-2007 Designer, Web-developer, 3D-modeler. Nika(Ukraine)
2007-2008 Designer, Web-developer, 3D-modeler. Freelance
2008-2009 Designer, Web-developer, 3D-modeler. Web-City(Ukraine)
2009-2014 Designer (predominantly Web-designer), Web-developer(HTML, CSS, JavaScript, ActionScript), 3D-modeler. Freelance
2014-2014 (6 month) Web-deweloper (Web-design, HTML5, CSS, JavaScript). WebMozaica(Ukraine)
2014-2016 Web-developer (rarely Web-design, sometimes HTML5(JADE), CSS(Sass),  predominantly JavaScript(LiveScript), NodeJS, AngularJS )

PERSONAL SKILLS 
Mother tongue(s) Ukrainian, Russian

Other language(s) English
Reading - Free
listening (clear conversation) - 75-85%
Lexicon - 2000-3000 words
Spoken interaction - not practiced
Spoken production - commenting code

Job-related skills 
▪ HTML/ CSS/ I began to use in 2006. Total experience  4-5 years 
▪ Jade/ Sass -  1 year
▪ JavaScript - Total experience  3-4 years
▪ CoffeScript(LiveScript) - 1 year
▪ C++ - 1 year
▪ ActionScript - 1 year
▪ Java - 2 month
▪ PHP — Rarely used to make changes to WordPress and Jumla and other CMS. Total experience 0.5 years 
▪ MySQL — Total experience 0.5 years 
▪ MongoDB - 6 month (current)

Frameworks: 
▪  AngularJS - 1year

Server Administration:
VPS + NodeJS - 1year

Graphics:
Photoshop: +10 year
Adobe Illustrator: +8year
3DSMax : +6year
CorelDrow: +4year

Intensive practice during study
▪  Paskal 3.0 - 7.0 - 3year
▪  Assembler - 8 month
▪  C++ 1 year

Many scattered experience:
Various CMS, programs. 
User experience - 19 years (Mainly for the development, design, training)

Other experience:
▪ Team lead
▪ Self-management projects
▪ Extensive experience of self-organization
▪ The organizational work
▪ Teaching
▪ Studied psychology - 17years
▪ Many years of studying and pondering: social systems, forms of social interaction, infrastructure systems

Works: 

www.adventurebucketlist.com/learnmore/ (Teamwork)
www.miprosco.com
northstar-limos.squarespace.com
rozvivalka.xyz
www.behance.net/spirit_urban

**Comment by robespier** [comment]
> ...Задача не из тривиальных, поэтому при выборе исполнителя в первую очередь мы будем ориентироваться на осмысленность вопросов.

**Ждем вопросов ;)**

**Comment by SpiritUrban** [comment]
Я поверхностно оценил. Теперь нужно осмыслить глубже. Потому что я пока в режиме поиска заказов. Сориентируйте уже конкретнее - на какой странице почитать задачу ? Потому что я уже ее потерял.

**Comment by SpiritUrban** [comment]
отписал там же )


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Без
вирусов. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

2016-06-11 23:21 GMT+03:00 Roman Sergeev <notifications@github.com>:

**Comment by robespier** [comment]
нам нужен Solver - модуль, формирующий массив трансформаций исходного контура.
Почитать можно здесь: https://github.com/robespier/IML_CutLayout/wiki/%D0%A7%D0%B5%D1%80%D0%BD%D1%8B%D0%B9-%D1%8F%D1%89%D0%B8%D0%BA-Solver

**Comment by SpiritUrban** [comment]
Чтобы понять суть описания нужно знать контекст. Это порезка лазером листа? Нужно разложить оптимально детали или учесть усадку? Или это просто модификатор (получаем одну фигуру и модефицируем в другую)? Несколько графических зарисовок, сразу бы прояснили ситуацию.

**Comment by SpiritUrban** [comment]
Ответил. Лучше вести переписку на гитхабе? Если да тогда пишем там. Но могу
втыкнуть. А почту полюбому не прогавлю.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Без
вирусов. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DDB4FAA8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

2016-06-11 23:46 GMT+03:00 Roman Sergeev <notifications@github.com>:

**Comment by SpiritUrban** [comment]
Из того что вижу - понимаю следующее. Нужно сделать оптимальную раскладку повторяющихся форм. Учесть процент горизонтального разреза, растояние (чтобы не образовалось слишком малых отступов), чтобы фигуре не пересекались. И нужно выбрать найболее плотное размещение из возможных.
 Интересно, почему нельзя много резать горизонтально.

**Comment by robespier** [comment]
> Чтобы понять суть описания нужно знать контекст. Это порезка лазером листа? Нужно разложить оптимально детали или учесть усадку? Или это просто модификатор (получаем одну фигуру и модефицируем в другую)? Несколько графических зарисовок, сразу бы прояснили ситуацию.

Нет, это не порезка лазером. Это высекательная машина Shober ротационного типа (можно погуглить). Нужно разложить делали. Примеры готовых раскладок можно посмотреть здесь: https://github.com/robespier/IML_CutLayout/blob/master/misc/Examples_layout_types.pdf
На входе - один контур, на выходе - массив трансформаций исходного контура (duplicate, rotate) и форма Области Размещения.

> Из того что вижу - понимаю следующее. Нужно сделать оптимальную раскладку повторяющихся форм. Учесть процент горизонтального разреза, растояние (чтобы не образовалось слишком малых отступов), чтобы фигуре не пересекались. И нужно выбрать найболее плотное размещение из возможных.

Да, примерно так. Только помимо плотного размещения необходимо соблюсти требования, в частности нужно избегать наложения элементов в ручьях ("столбцах таблицы" скажем так).

**Comment by SpiritUrban** [comment]
В общем я понял. Это нужно садится и решать задачу. Причем промежуточный результат здесь вряд ли возможен. Из этого выплывает следующий формат работы: не решил, нет гонорара, решил - вероятно гонорар будет + возможно дальнейшее сотрудничество.

**Comment by robespier** [comment]
все верно.

**Comment by robespier** [comment]
по поводу садится и решать - мы поможем. исходные данные в формате json предоставим и будем тестировать выходные данные и вносить коррективы, какие надо. по деньгам - обсуждаемо. бюджет не резиновый, но договориться можно. причем, желательно оценить свою работу и озвучить сумму до того, как садиться и решать задачу, ну или хотя бы в процессе решения.
